moha/hamiltonians.py

```diff
@@ -124,10 +126,23 @@ def generate_one_body_integral(self, basis: str, dense: bool):
         -------
         scipy.sparse.csr_matrix or np.ndarray
         """
-        one_body_term = (
+        
+        # check if all parameters are integers in connectivity matrix
+        if isinstance(self.connectivity, np.ndarray):
+            one_body_term = (
                 diags([self.alpha for _ in range(self.n_sites)], format="csr")
                 + self.beta * self.connectivity_matrix
-        )
+            )
+        elif np.all([isinstance(k, int) for _, _, k in self.connectivity]):
+            one_body_term = assign_rauk_parameters(self.connectivity, self.atom_types, 
```

Comment:
I think it `assign_rauk_parameters` function should be able to take just a connectivity and return one body term. It shouldn't need atom types, atoms_nums and number of sites. This should all be parsed from connectivity